cli/cmdlineparser.cpp

```diff
@@ -1405,9 +1405,12 @@ CmdLineParser::Result CmdLineParser::parseFromArgs(int argc, const char* const a
     else if ((def || mSettings.preprocessOnly) && !maxconfigs)
         mSettings.maxConfigs = 1U;
 
-    if (mSettings.checks.isEnabled(Checks::unusedFunction) && mSettings.jobs > 1 && mSettings.buildDir.empty()) {
-        // TODO: bail out
-        mLogger.printMessage("unusedFunction check can't be used with '-j' option. Disabling unusedFunction check.");
+    if (mSettings.jobs > 1 && mSettings.buildDir.empty()) {
+        // TODO: bail out instead?
+        if (mSettings.checks.isEnabled(Checks::unusedFunction))
+            mLogger.printMessage("unusedFunction check requires --cppcheck-build-dir to be active with -j.");
+        // TODO: enable
+        //mLogger.printMessage("whole program analysis requires --cppcheck-build-dir to be active with -j.");
```

Comment:
I did not enable this since it causes too much noise by default. But essentially it is not different from the above message and it is also rather important. Can probably be enabled after we re-worked the message output.

Reply:
for information I would like to investigate someday to fix this.

Reply:
If all the shortcomings are finally fixed I might take a look at that as well.